### PR TITLE
Feature/refactor database pathrefs

### PR DESCRIPTION
**Pull request by NathanDVT**
- [X] Restructured database design
- [X] Refactored repo method calls
- [X] Refactored fire database calls to support new structure
- [X] Modified DTO structures to meet screen requirements

image

### CODE REVIEW

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/NLibrary/pullRequest?prid=5415919&bid=17756886)

File: NLibrary/Classes/PlaylistMVVM/PlaylistModel.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+public struct PlaylistModel {
+    public var playlistId: String = ""
+    public var playlistName: String = ""
+    public var producerEmail: String = ""
+    public var producerId: String = ""
+    public var numSongs: Int = 0
+    public var basicSongItems: [BasicSongItem] = []
+
```
Comment (NathanDVT) [inline NLibrary/Classes/PlaylistMVVM/PlaylistModel.swift:39]: Codacy Issue found: [Lines should not have trailing whitespace.](https://app.codacy.com/manual/NathanDVT/NLibrary/pullRequest?prid=5415919)

File: NLibrary/Classes/PlaylistMVVM/PlaylistModel.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+//            guard let key = key as? String,
+//                let page = value as? NSDictionary else {
+//                return
+//            }
+//            playlistItems.append(PlaylistModel(albumName: key, dictionary: page))
+//        }
+//    }
+//    var dict: [String: Any]?
+}
+
```
Comment (XanderSchoeman) [inline NLibrary/Classes/PlaylistMVVM/PlaylistModel.swift]: Maybe just remove the commented code if it's not being used.

**Review by XanderSchoeman — CHANGES_REQUESTED** [review]
gotta get that Codacy standards status check to pass ; )

File: NLibrary/Classes/PlaylistMVVM/PlaylistModel.swift (hunk 2 above)
Comment (NathanDVT) [reply NLibrary/Classes/PlaylistMVVM/PlaylistModel.swift]: Thanks will do so now

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/NLibrary/pullRequest?prid=5415919&bid=17756886)

File: NLibrary/Classes/PlaylistMVVM/PlaylistModel.swift [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import AVKit
+
+public struct PlaylistModel {
+    public var playlistId: String = ""
+    public var playlistName: String = ""
+    public var producerEmail: String = ""
+    public var producerId: String = ""
+    public var numSongs: Int = 0
+    public var basicSongItems: [BasicSongItem] = []
+
```
Comment (NathanDVT) [inline NLibrary/Classes/PlaylistMVVM/PlaylistModel.swift]: Codacy Issue found: [Lines should not have trailing whitespace.](https://app.codacy.com/manual/NathanDVT/NLibrary/pullRequest?prid=5415919)

**Review by XanderSchoeman — APPROVED** [review]

**Pull request merged**